pkg/machine/hyperv/config.go

```diff
@@ -0,0 +1,284 @@
+	if err != nil {
+		return nil, err
+	}
+
+	configPath, err := machine.NewMachineFile(getVMConfigPath(configDir, opts.Name), nil)
+	if err != nil {
+		return nil, err
+	}
+	m.ConfigPath = *configPath
+
+	ignitionPath, err := machine.NewMachineFile(filepath.Join(configDir, m.Name)+".ign", nil)
+	if err != nil {
+		return nil, err
+	}
+	m.IgnitionFile = *ignitionPath
+
+	// Set creation time
+	m.Created = time.Now()
+
+	dataDir, err := machine.GetDataDir(machine.HyperVVirt)
+	if err != nil {
+		return nil, err
+	}
+
+	// Acquire the image
+	// Until we are producing vhdx images in fcos, all images must be fed to us
+	// with --image-path.  We should, however, accept both a file or url
+	g, err := machine.NewGenericDownloader(machine.HyperVVirt, opts.Name, opts.ImagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	imagePath, err := machine.NewMachineFile(g.Get().GetLocalUncompressedFile(dataDir), nil)
+	if err != nil {
+		return nil, err
+	}
+	m.ImagePath = *imagePath
+	if err := machine.DownloadImage(g); err != nil {
+		return nil, err
+	}
+
+	config := hypervctl.HardwareConfig{
+		CPUs:     uint16(opts.CPUS),
+		DiskPath: imagePath.GetPath(),
+		// TODO remove this and make real
+		DiskSize: 100,
+		Memory:   int32(opts.Memory),
+	}
+
+	// Write the json configuration file which will be loaded by
+	// LoadByName
+	b, err := json.MarshalIndent(m, "", " ")
+	if err != nil {
+		return nil, err
+	}
+	if err := os.WriteFile(m.ConfigPath.GetPath(), b, 0644); err != nil {
+		return nil, err
+	}
+
+	vmm := hypervctl.NewVirtualMachineManager()
+	if err := vmm.NewVirtualMachine(opts.Name, &config); err != nil {
+		return nil, err
+	}
+	return v.LoadVMByName(opts.Name)
+
+}
+
+func (v Virtualization) RemoveAndCleanMachines() error {
+	// Error handling used here is following what qemu did
+	var (
+		prevErr error
+	)
+
+	// The next three info lookups must succeed or we return
+	mms, err := v.loadFromLocalJson()
+	if err != nil {
+		return err
+	}
+
+	configDir, err := machine.GetConfDir(vmtype)
+	if err != nil {
+		return err
+	}
+
+	dataDir, err := machine.GetDataDir(vmtype)
+	if err != nil {
+		return err
+	}
+
+	vmm := hypervctl.NewVirtualMachineManager()
+	for _, mm := range mms {
+		vm, err := vmm.GetMachine(mm.Name)
+		if err != nil {
+			prevErr = handlePrevError(err, prevErr)
+		}
+
+		// If the VM is not stopped, we need to stop it
+		// TODO stop might not be enough if the state is dorked. we may need
+		// something like forceoff hard switch
+		if vm.State() != hypervctl.Disabled {
+			if err := vm.Stop(); err != nil {
+				prevErr = handlePrevError(err, prevErr)
+			}
+		}
+		if err := vm.Remove(mm.ImagePath.GetPath()); err != nil {
+			prevErr = handlePrevError(err, prevErr)
+		}
+	}
+
+	// Nuke the config and dataDirs
+	if err := os.RemoveAll(configDir); err != nil {
+		prevErr = handlePrevError(err, prevErr)
+	}
+	if err := os.RemoveAll(dataDir); err != nil {
+		prevErr = handlePrevError(err, prevErr)
+	}
```

Comment:
Does removing configDir remove just this Machine's config, or all machines?

Reply:
all